### PR TITLE
remove dynamicrestmapper which is deprecated

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**

Remove `github.com/operator-framework/operator-sdk/pkg/restmapper` which was deprecated in the version `0.14.0` in favor of the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). ([#2544](https://github.com/operator-framework/operator-sdk/pull/2544))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 ### Removed
 
+- Remove `github.com/operator-framework/operator-sdk/pkg/restmapper` which was deprecated in the version `0.14.0` in favor of the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). ([#2544](https://github.com/operator-framework/operator-sdk/pull/2544))
```
Comment (estroz) [inline CHANGELOG.md]:
```suggestion
- **Breaking Change:** remove `pkg/restmapper` which was deprecated in `v0.14.0`. Projects that use this package must switch to the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). ([#2544](https://github.com/operator-framework/operator-sdk/pull/2544))
```

**Comment by camilamacedo86** [comment]
Hi @estroz, has any reason for we need to wait for wait until v0.17.2?
I do not think that it has relation with.

**Comment by joelanford** [comment]
@estroz It was deprecated in v0.14.x and v0.15.x, so I think we can remove in v0.16.x.

However, maybe since we're just now getting around to writing up the migration guides for v0.13, v0.14, and v0.15, we should hold off until v0.17.x since some users may not yet know about the deprecation.

**Comment by camilamacedo86** [comment]
Close for now, and re-open when we can merge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 ### Removed
 
+- **Breaking Change:** remove `pkg/restmapper` which was deprecated in `v0.14.0`. Projects that use this package must switch to the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDiscoveryRESTMapper). ([#2544](https://github.com/operator-framework/operator-sdk/pull/2544))
```
Comment (estroz) [inline CHANGELOG.md]:
```suggestion
- **Breaking Change:** remove `pkg/restmapper` which was deprecated in `v0.14.0`. Projects that use this package must switch to the `DynamicRESTMapper` implementation in [controller-runtime](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/apiutil#NewDynamicRESTMapper). ([#2544](https://github.com/operator-framework/operator-sdk/pull/2544))
```

**Pull request merged**